libraries/extractor/src/main/java/androidx/media3/extractor/text/ssa/SsaParser.java

```diff
@@ -461,11 +426,104 @@ private static Cue createCue(
       cue.setPosition(styleOverrides.position.x / screenWidth);
       cue.setLine(styleOverrides.position.y / screenHeight, LINE_TYPE_FRACTION);
     } else {
-      // TODO: Read the MarginL, MarginR and MarginV values from the Style & Dialogue lines.
       cue.setPosition(computeDefaultLineOrPosition(cue.getPositionAnchor()));
       cue.setLine(computeDefaultLineOrPosition(cue.getLineAnchor()), LINE_TYPE_FRACTION);
     }
 
+    // Apply margins if there are no overrides and we have valid positions.
+    if (styleOverrides.alignment == SsaStyle.SSA_ALIGNMENT_UNKNOWN
```

Comment:
why do we check specifically `styleOverrides.alignment` here when we've already resolved `alignment` from both `styleOverrides` and `style` on L411?

(related comment below)